File: spring_boot/biblioteca_fullstack/biblioteca-somente-java/src/main/java/ubsocial/com/biblioteca/model/entity/Livro.java
package ubsocial.com.biblioteca.model.entity;

import java.util.Date;
import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

@Entity
public class Livro {

    public Livro() {
    }

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @NotBlank
    @Size(min = 1, max = 255, message = "Título precisa ter entre 1 e 255 caracteres")
    @Column(length = 255)
    private String titulo;

    @NotBlank
    @Size(min = 1, max = 255, message = "Autor precisa ter entre 1 e 255 caracteres")
    @Column(length = 255)
    private String autor;

    @NotNull
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    @Temporal(TemporalType.DATE)
    private Date dataPublicacao;

    @NotBlank
    @Column(unique = true)
    @Size(min = 13, max = 13, message = "ISBN precisa ter 13 caracteres")
    private String isbn;

    private int paginas;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getAutor() {
        return autor;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public Date getDataPublicacao() {
        return dataPublicacao;
    }

    public void setDataPublicacao(Date dataPublicacao) {
        this.dataPublicacao = dataPublicacao;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public int getPaginas() {
        return paginas;
    }

    public void setPaginas(int paginas) {
        this.paginas = paginas;
    }
}

File: spring_boot/biblioteca_fullstack/biblioteca/src/main/kotlin/ubsocial/com/biblioteca/model/entity/Livro.kt
package ubsocial.com.biblioteca.model.entity
import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import java.time.LocalDate

@Entity
data class Livro(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: Long? = null,

    @field:NotBlank
    @field:Size(min = 1, max = 255, message = "Título precisa ter entre 1 e 255 caracteres")
    @Column(length = 255)
    val title: String = "",

    @field:NotBlank
    @field:Size(min = 1, max = 255, message = "Autor precisa ter entre 1 e 255 caracteres")
    @Column(length = 255)
    val author: String = "",

    @field:NotNull
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    @Column(name = "published_date")
    val published_date: LocalDate = LocalDate.now(),

    @field:NotBlank
    @field:Size(min = 13, max = 13, message = "ISBN precisa ter 13 caracteres")
    @Column(unique = true)
    val isbn: String = "",

    val pages: Int = 0,

    @field:Size(min = 1, max = 255, message = "Capa/Gênero precisa ter entre 1 e 255 caracteres")
    @Column(length = 255)
    val cover: String? = null,

    @field:NotBlank
    @field:Size(min = 1, max = 255, message = "Idioma precisa ter entre 1 e 255 caracteres")
    @Column(length = 255)
    val language: String = ""
)